### PR TITLE
Fix delete user

### LINKED ISSUE #110: Fix deleteRallyUser race condition
Upon account deletion, we first go and delete all the studies in the user's subcollection. Those deletions will at some point trigger `handleUserStudyChange` , and it's very possible that the user's `extensionUser` document is deleted before `handleUserStudyChange` is run, since the trigger might not happen immediately and the delete function will still continue on its merry way.

Since this is deletion we are talking about, I can just make the `handleUserStudyChange` function a no-op if the study is being deleted, since the overall user "unenrollment" ping will already be sent i.e. we do not need to send individual "studyUnenrollment" pings for each study in this case.

### LINKED ISSUE #111: Delete extension-prefixed auth accounts when corresponding user is deleted
Right now when a user deletes their account, we don't delete the other auth accounts that we generated for each study they joined. This leaves a bunch of unnecessary auth accounts lying around, and we should delete them.

### PR DESCRIPTION
**Pull request by aaga**
Closes #110 
Closes #111 

Reviewer expectations:
- Confirm functionality by testing on Stage or in your own environment
- If someone is able to manually test with multiple studies, that would be awesome
- @rhelmer can you test with our current Svelte site?
- @thomik-corp can you test with the new React site?
- Urgency: normal

### CODE REVIEW

File: web-platform/functions/src/index.ts [hunk 1]
```diff
@@ -135,6 +144,10 @@ export async function deleteRallyUserImpl(
   // Do a final commit in case we ended on a partial batch.
   await batch.commit();
 
+  // Delete all the study-specific auth accounts associated with this user
+  // Limited to 1000, but it is safe to assume we will not have more than 1000 studies
```
Comment (rhelmer) [inline web-platform/functions/src/index.ts:148]: You're probably right, what I did elsewhere was to do this sort of operation in batches. We could probably stand to have some helper functions that make it easier to do this, then we don't need to worry about any hidden limits. Would you mind filing an issue to follow-up? I think you're right that worrying this in the near-term is not very useful, but I'm concerned that these may start to balloon if we expand what the definition of a "study" is and it's not 1:1 with extensions anymore.

**Review by rhelmer — CHANGES_REQUESTED** [review]
Code lgtm, please file a followup on making it easier to do this stuff in batches (or if you don't think it's every going to be worthwhile that's fine too). I'll test this too per your review expectations.

File: web-platform/functions/src/index.ts (hunk 1 above)
Comment (aaga) [reply web-platform/functions/src/index.ts:148]: Mhmm yes I saw the batching for deleting the user's study docs further up in this function.

I think even if we have way more studies, this line here is purely for deleting the user accounts associated with the extension, and I can't imagine we'd have more than 1000 extensions(!!). But it is a good point if for example we decide to start creating a new user account for each study, regardless of if it has an extension or not.

Just filed an issue for it: https://github.com/mozilla-rally/rally/issues/153

**Review by rhelmer — APPROVED** [review]

**Review by rhelmer — APPROVED** [review]
This works in my local testing of the emulator + Svelte app, thanks!

**Review by thomik-corp — APPROVED** [review]

**Pull request merged**